Revit_Core_Engine/Convert/Environment/FromRevit/CableTray.cs

```diff
@@ -69,7 +69,17 @@ public static List<BH.oM.MEP.System.CableTray> CableTrayFromRevit(this Autodesk.
             // Orientation angle
             double orientationAngle = revitCableTray.OrientationAngle(settings);
 
-            List<BH.oM.Geometry.Line> queried = Query.LocationCurveMEP(revitCableTray, settings);
+            bool isStartConnected = false;
+            bool isEndConnected = false;
+            List<BH.oM.Geometry.Line> queried = Query.LocationCurveMEP(revitCableTray, out isStartConnected, out isEndConnected, settings);
+            Vector revitCableTrayVector = BH.Engine.Geometry.Modify.RoundCoordinates(
+                VectorFromRevit((revitCableTray.Location as LocationCurve).Curve.GetEndPoint(0) -
+                                (revitCableTray.Location as LocationCurve).Curve.GetEndPoint(1)),4).Normalise();
```

Comment:
Could we please avoid splitting a single statement into 3 lines? 

I would also consider rewriting the query to make the code a bit more readable:
```suggestion
            Vector revitCableTrayVector = ((revitCableTray.Location as LocationCurve)?.Curve as Line)?.Direction.VectorFromRevit().RoundCoordinates(4);
```

On top of that, this variable is not used until line 84, so could be declared later.